server/server.go

```
@@ -0,0 +1,238 @@
+	*BaseServer,
+	error,
+) {
+	if attester == nil {
+		return nil, errors.New("attester cannot be nil")
+	}
+	if tlsBundle == nil {
+		return nil, errors.New("tlsBundle cannot be nil")
+	}
+	tlsConfig, err := tlsBundle.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	router := chi.NewRouter()
+
+	srv := &BaseServer{
+		appURL:   appURL,
+		appPort:  appPort,
+		attester: attester,
+		cert:     tlsBundle.GetCert, // as function since not known until later
+		httpSrv: http.Server{
+			Addr:      fmt.Sprintf(":%d", appPort),
+			Handler:   router,
+			TLSConfig: tlsConfig,
+		},
+		inEnclave: inEnclave,
+		router:    router,
+	}
+
+	return srv, nil
+}
+
+func NewPartialBaseServerFromRaw(
+	attester attestation.Attester,
+	cert func(ctx context.Context) (certificate.Cert, error),
+) *BaseServer {
+	return &BaseServer{
+		attester: attester,
+		cert:     cert,
+	}
+}
+
+func (server *BaseServer) InEnclave() bool {
+	return server.inEnclave
+}
+
+func (server *BaseServer) AppURL() string {
+	return server.appURL
+}
+
+func startError(msg string, err error) error {
+	return fmt.Errorf("failed to start Server: %v: %w", msg, err)
+}
+
+func (server *BaseServer) Start() error {
+	if server.inEnclave {
+		if err := system.AssignLoAddr(); err != nil {
+			return startError("failed to assign loopback address", err)
+		}
+		elog.Println("Assigned address to lo interface.")
+	}
+
+	// Set file descriptor limit. There's no need to exit if this fails.
+	err := system.SetFdLimit(system.DefaultFdSoft, system.DefaultFdHard)
+	if err != nil {
+		elog.Print(
+			startError("failed to set file descriptor limit", err).
+				Error(),
+		)
+	}
+
+	// Start the web server.  If we're inside an enclave, we use a
+	// vsock-enabled listener, otherwise a simple tcp listener.
+	if server.inEnclave {
+		listener, err := vsock.Listen(uint32(server.appPort), nil)
+		if err != nil {
+			return startError("failed to create vsock listener", err)
+		}
+		defer func() {
+			_ = listener.Close()
```

I would suggest not squashing errors.  If nothing else, consider logging.

---

done